libraries/ms-helper/inc/thermistor.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+// Thermistor Reading interface for NXRT15XH103FA1B (10k Ohm NTC)
+// Requires ADC, GPIO and Interrupts to be initialized.
+//
+// The thermistor and its fixed resistor forms a voltage divider.
+// A temperature value ranging from 0~100 degrees is calculated from the voltage divider.
+#include "adc.h"
+#include "gpio.h"
+
+// A 10kOhm resistor is used as the fixed value
+#define THERMISTOR_FIXED_RESISTANCE_OHMS 10000
+
+// ThermistorPosition indicates which resistor the node voltage is measured from
+// ex: VDDA ---> R1 ---> R2 ---> GND
+typedef enum {
+  THERMISTOR_POSITION_R1 = 0,
+  THERMISTOR_POSITION_R2,
+  NUM_THERMISTOR_POSITIONS,
+} ThermistorPosition;
+
+typedef struct ThermistorStorage {
+  ThermistorPosition position;
+  ADCChannel adc_channel;
+} ThermistorStorage;
+
+// Initializes the GPIO pin and ADC Channel associated with the thermistor
+StatusCode thermistor_init(ThermistorStorage *storage, GPIOAddress thermistor_gpio,
+                           ThermistorPosition position);
+
+// Fetch the temperature reading in milliCelsius from the MCU's ADC
+StatusCode thermistor_get_temp(ThermistorStorage *storage, uint32_t *temperature_millicelcius);
+
+// Calculate the temperature in milliCelsius from milliohms
+StatusCode thermistor_calculate_temp(uint32_t thermistor_resistance_milliohms,
```

Comment:
I feel like ohms is probably good enough for the input. If you want to store mOhms internally that's fine, but we're probably good with ohms when calculating the temperature.

Similarly, I doubt that the result we're getting in mC is actually that precise. It's probably good enough to just use 0.1C units. It'll just make using the module a little easier on the brain.

Reply:
I don't feel like its necessary to change the resistance input to Ohms. The lookup table converges quickly onto one value when it approaches high temperatures. Using ohms for the input would remove data that could otherwise make a great impact. Nor do I think the resistance would be hard to use as the name itself is pretty self-explanatory.

Reply:
I do agree with the celsius changes; millicelsius is asinine levels of precision